meshtastic/tests/test_node.py

```
@@ -669,6 +669,78 @@ def test_getChannelByChannelIndex():
+@pytest.mark.unit
+def test_delete_channel_rejects_primary():
+    """deleteChannel should refuse deleting PRIMARY slot 0."""
+    iface = MagicMock()
+    anode = Node(iface, "bar", noProto=True)
+    iface.localNode = anode
+    anode.channels = _build_channels(3)
+
+    with pytest.raises(SystemExit) as pytest_wrapped_e:
+        anode.deleteChannel(0)
+
+    assert pytest_wrapped_e.type == SystemExit
```

Ruff (0.15.20)

[error] Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)